packages/devtools/lib/src/timeline/timeline_protocol.dart

```diff
@@ -324,7 +324,7 @@ class TimelineData {
 
     previousDurationEndEvents[event.type.index] = event;
 
-    current.endTime = event.timestampMicros;
+    current.timeRange.end = Duration(microseconds: event.timestampMicros);
```

Comment:
generally it is cleaner to make these sorts of trivial data classes immutable.  That way is you use one as a key in a Map things will work instead of going haywire if values change.